### PR TITLE
Added setuptools to make blockdiag work

### PR DESCRIPTION
**Pull request by fnchooft**
Trying to generate some nwdiag, etc i had to install python setuptools. The adapted Dockerfile should fix this.

### CODE REVIEW

**Comment by dduportal** [comment]
Hello @fnchooft , thank you for your contribution.

Could you update the test harness with your use case?

Also, what is the size difference this package is introducing?

**Comment by fnchooft** [comment]
Hello @dduportal , 
size: https://pkgs.alpinelinux.org/package/edge/main/x86/py-setuptools - installed size seems around 2.5 MB

```
bash-4.3# apk info py-setuptools
py-setuptools-33.1.1-r1 description:
A collection of enhancements to the Python distutils
py-setuptools-33.1.1-r1 webpage:
http://pypi.python.org/pypi/setuptools
py-setuptools-33.1.1-r1 installed size:
2609152
```
Will add a test.
Kind regards,
Fabian

**Comment by dduportal** [comment]
Thanks @fnchooft for the size information: seems acceptable.

In order to get the issue fixed, we would need you to provide a test case, or at least a document example, with the expected result, and the result you currently have.

This is important to ensure the fix that will be merged to master:
* Is the right fix (maybe we only need nwdiag installed, so setup-tools could be removed from the final image, as the build-base package is
* Won't come again in the future
* Is becoming part of "explicit and shared knowledge" as part of the test harness

Thanks!

**Comment by jirutka** [comment]
setuptools package is named `py-setuptools`, not `py2-setuptools`. We have not renamed it yet.

BTW, @dduportal, why do you use Legacy Python instead of Python 3 in the image? Some of the used tools is not compatible with recent Python yet?

**Comment by fnchooft** [comment]
@jirutka: You are right my apologies, it is py-setuptools. @dduportal: I am not familiar enough with python to give my opinion just yet.  py-setuptools does not seem to be required by blockdiag as per dependencies ( https://pkgs.alpinelinux.org/package/edge/testing/x86/py-blockdiag ) , neither nwdiag,etc.

**Comment by dduportal** [comment]
Hello @fnchooft , I am confused. But the test case might solve this confusion: whats is the initial issue you have and want to solve? I mean: installing setuptools for "just installing setup tools" is not a goal. But having an Asciidoctor document with "nwdiag" used for rendering a diagram, and facing an error or an empty image, while expecting something else is a good way to start.

This is why I'm asking for a test case, even just as a comment here (and I'll help on implementing this with pleasure!) , before applying or merging a solved case.

@jirutka thanks fro pointing this to me. I think I did not pay attention when the `py2-pillow` packages had been added, and messed myself here. 
TL;DR; There is no reason to stick on python 3. Just opened #59 for this.

**Comment by fnchooft** [comment]
@dduportal : I added additional drawings to the basic-drawing example which were not passing without the py-setuptools. Let me know if you need anything else.
Kind regards,
Fabian

**Comment by fnchooft** [comment]
Some more background information:
```
bash-4.3# asciidoctor -r asciidoctor-diagram diagrams.ad 
asciidoctor-diagram: ERROR: Failed to generate image: nwdiag failed: Traceback (most recent call last):
  File "/usr/bin/nwdiag", line 7, in <module>
    from nwdiag.command import main
  File "/usr/lib/python2.7/site-packages/nwdiag/command.py", line 18, in <module>
    import nwdiag.builder
  File "/usr/lib/python2.7/site-packages/nwdiag/builder.py", line 18, in <module>
    from nwdiag.elements import (Diagram, DiagramNode, DiagramEdge,
  File "/usr/lib/python2.7/site-packages/nwdiag/elements.py", line 17, in <module>
    import blockdiag.elements
  File "/usr/lib/python2.7/site-packages/blockdiag/elements.py", line 22, in <module>
    from blockdiag import noderenderer, plugins
  File "/usr/lib/python2.7/site-packages/blockdiag/noderenderer/__init__.py", line 17, in <module>
    import pkg_resources
ImportError: No module named pkg_resources
```

The missing dependency is *pkg_resources*.
http://svn.python.org/projects/sandbox/branches/setuptools-0.6/pkg_resources.txt

Since it is distributed with py_setuptools - i did the following inside the docker-image:
```
bash-4.3# apk add --no-cache py-setuptools
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
(1/1) Installing py-setuptools (33.1.1-r1)
Executing busybox-1.26.2-r7.trigger
OK: 235 MiB in 100 packages
```

After which: 
```
bash-4.3# asciidoctor -r asciidoctor-diagram diagrams.ad 
```
Runs without problem.

**Comment by dduportal** [comment]
Thanks for the details @fnchooft.

I will pull request your repository with the tests integrated in the automatic harness, unless you want  to do it yourself (in that case, I let you check the other tests in the `tests` directory.

I won't be able to work on this right now, but expect something before the end of the week.

File: tests/fixtures/sample-with-diagram.adoc [hunk 1]
```diff
@@ -14,3 +14,119 @@ Thanks to link:https://github.com/ryosms[]!
 class Sample {
 }
 ----
+
+
+= Additional Diagrams
```
Comment (dduportal) [inline tests/fixtures/sample-with-diagram.adoc]: Can you remove this line ? It is giving this error when running manually
```
bash-4.3# asciidoctor -r asciidoctor-diagram sample-with-diagram.adoc 
asciidoctor: ERROR: sample-with-diagram.adoc: line 19: only book doctypes can contain level 0 sections
```

File: tests/fixtures/sample-with-diagram.adoc [hunk 2]
```diff
@@ -14,3 +14,119 @@ Thanks to link:https://github.com/ryosms[]!
...
+    a -> b
+    b -> c
+    c -> d
+    d -> a
+}
+----
+
+== Plantuml from external file...
+
+plantuml::activity_diagram.txt[format="svg", align="center"]
```
Comment (dduportal) [inline tests/fixtures/sample-with-diagram.adoc]: It seems there is a bug on the macro that hit us here (relative path is resolved from the caller and not from the document): https://github.com/asciidoctor/asciidoctor-diagram/issues/173

Can you use the block + include syntax here, since it is not related to the docker image ?

Here is a sample I tried locally with success:
```
[plantuml,activity,svg,align="center"]
----
include::activity_diagram.txt[]
----
```

File: tests/fixtures/sample-with-diagram.adoc (hunk 1 above)
Comment (fnchooft) [reply tests/fixtures/sample-with-diagram.adoc]: I could make it a == instead of =. 
That way the doctype is satisfied.

File: tests/fixtures/sample-with-diagram.adoc (hunk 2 above)
Comment (fnchooft) [reply tests/fixtures/sample-with-diagram.adoc]: Ok - will do.

File: tests/fixtures/sample-with-diagram.adoc (hunk 1 above)
Comment (dduportal) [reply tests/fixtures/sample-with-diagram.adoc]: @fnchooft it would be great. Thank you!

**Comment by dduportal** [comment]
@fnchooft I wish you an happy new year. Thanks for your feedback and precious time spend on this contribution!

**Comment by fnchooft** [comment]
@dduportal - Thanks for merging.
Have a productive 2018!
Kind regards,
Fabian
